### PR TITLE
fix the error of GH104

### PR DESCRIPTION
**Pull request by tkoyama010**
### Description

See https://learn.scientific-python.org/development/guides/gha-wheels#GH104.
See #684.

---

### CODE REVIEW

**Comment by tkoyama010** [comment]
@bjlittle OK. I fixed the new GH104 rules. It is a good rule in my mind.

**Review by bjlittle — CHANGES_REQUESTED** [review]
@tkoyama010 Brilliant thanks!

After reading through the `actions/download-artifact` docs, I realised that they've bumped to `v4.1.1`, which also includes fixes in `v4.1.0` that allows multiple artifacts to be merged (https://github.com/actions/download-artifact/pull/259). This is exactly the behaviour we want :+1: 

So we can finally unpin from `v3` of `actions/upload-artifact` and `actions/download-artifact` and use the new `merge-multiple` support (see the [README Usage](https://github.com/actions/download-artifact/pull/259) section and example patterns).

As an aside, note that there is now a new [@actions/upload-artifact/merge](https://github.com/actions/upload-artifact/blob/main/merge/README.md) GHA :star_struck:

File: .github/workflows/ci-wheels.yml [hunk 1]
```diff
@@ -84,7 +84,7 @@ jobs:
 
     - uses: actions/download-artifact@v3
       with:
-        name: pypi-artifacts
+        name: pypi-artifacts-build-artifacts-0
```
[the remark this answers is not in the document]
Comment (tkoyama010) [reply .github/workflows/ci-wheels.yml:87]: Awesome expression!

**Review by bjlittle — APPROVED** [review]

**Comment by bjlittle** [comment]
@tkoyama010 Awesome, thanks for being all over this! :beers: :rocket:

**Pull request merged**